components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/handler/impl/BFSIDefaultResponseTypeHandlerImpl.java

```diff
@@ -0,0 +1,171 @@
+
+    /**
+     * return the new refresh validity period.
+     *
+     * @param oAuthAuthzReqMessageContext
+     * @return
+     */
+    public long updateRefreshTokenValidityPeriod(OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext) {
+
+        return oAuthAuthzReqMessageContext.getRefreshTokenvalidityPeriod();
+    }
+
+    /**
+     * return the new approved scope.
+     *
+     * @param oAuthAuthzReqMessageContext
+     * @return
+     */
+    public String[] updateApprovedScopes(OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext) {
+
+        if (oAuthAuthzReqMessageContext != null && oAuthAuthzReqMessageContext.getAuthorizationReqDTO() != null) {
+
+            String[] scopes = oAuthAuthzReqMessageContext.getApprovedScope();
+            if (scopes != null && !Arrays.asList(scopes).contains("api_store")) {
+
+                String sessionDataKey = oAuthAuthzReqMessageContext.getAuthorizationReqDTO().getSessionDataKey();
+                String consentID = getConsentIDFromSessionData(sessionDataKey);
+                if (consentID.isEmpty()) {
+                    log.error("Consent-ID retrieved from request object claims is empty");
+                    return scopes;
+                }
+
+                String consentIdClaim = IdentityExtensionsDataHolder.getInstance().getConfigurationMap()
+                        .get(ConsentManagementConstants.CONSENT_ID_CLAIM_NAME).toString();
+                String consentScope = consentIdClaim + consentID;
+                if (!Arrays.asList(scopes).contains(consentScope)) {
+                    String[] updatedScopes = ArrayUtils.addAll(scopes, consentScope);
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("Updated scopes: %s", Arrays.toString(updatedScopes)
+                                .replaceAll("[\r\n]", "")));
+                    }
+                    return updatedScopes;
+                }
+            }
+
+        } else {
+            return new String[0];
+        }
+
+        return oAuthAuthzReqMessageContext.getApprovedScope();
+    }
+
+    /**
+     * Call sessionDataAPI and retrieve request object, decode it and return consentID.
+     *
+     * @param sessionDataKey sessionDataKeyConsent parameter from authorize request
+     * @return consentID
+     */
+    String getConsentIDFromSessionData(String sessionDataKey) {
+
+        String consentID = StringUtils.EMPTY;
+        if (sessionDataKey != null && !sessionDataKey.isEmpty()) {
+            RequestObjectService requestObjectService = IdentityExtensionsDataHolder.getInstance()
+                    .getRequestObjectService();
+            if (requestObjectService != null) {
+                consentID = retrieveConsentIDFromReqObjService(requestObjectService, sessionDataKey);
+                if (consentID.isEmpty()) {
+                    log.error("Failed to retrieve ConsentID from query parameters");
```

Comment:
Are we terminating the flow when the ```consentID``` is empty from an upper level? If not, shouldn't this and other error logs be warn logs?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/884f24f4261bc92a886a68a4e8a5bf56e7a61136